internal/deepdoc/parser/pdf/parser_ocr_warp_test.go

```
@@ -0,0 +1,132 @@
+}
+
+// TestOCRDetectAndRecognize_WarpsCrop locks that ocrDetectAndRecognize feeds
+// the perspective-de-skewed (WarpCrop) crop to recognition, not the old
+// axis-aligned FastCrop of the detection bbox. This is the live-path wiring
+// of Step 4 / layer 1.
+func TestOCRDetectAndRecognize_WarpsCrop(t *testing.T) {
+	p := newTestParser()
+	page := image.NewRGBA(image.Rect(0, 0, 200, 140))
+
+	// A clearly skewed (perspective) quad: TL, TR, BR, BL.
+	quad := [4]util.Pt{{50, 40}, {150, 30}, {160, 90}, {40, 100}}
```

golangci-lint (2.12.2)

[error] composites: ragflow/internal/deepdoc/parser/pdf/util.Pt struct literal uses unkeyed fields

(govet)